### DISCUSSION TITLE
Network Requests

### DISCUSSION THREAD

**Discussion by paulzhdanov** [RavenSystem/esp-homekit-devices#1858] [Show and tell]
@RavenSystem please help: I took the code from your [example here](https://github.com/RavenSystem/esp-homekit-devices/wiki/Accessory-Configuration#send-httptcp-request-example-2), v10.8.3 Kestrel, but I'm getting empty POST on the backend. In the same time CURL emulations works perfect. Could you take a look if anything' wrong?

**SOLUTION: add header parameter** 
`"e": "Content-type: application/x-www-form-urlencoded\r\n", `

```
{
  "c":{"b":[{"g":0,"t":5}]},
  "a": [{
    "t":24,
    "g":5,
    "j":1,
    "y0": [{
      "v": 0,
      "r": 1,
      "0": { "h": [{ "h": "192.168.1.101", "m":2, "u": "write2.php", "c": "temperature=#HAA@0000" }] }
    }]
  }]
}

```

**Comment by RavenSystem** [comment]
Try to use a greater interval, like "j":10.

**Reply by paulzhdanov** [reply]
It didn't help, POST array is empty inside the script on backend. In the same time the code below works perfect:
`curl -X POST 'http://192.168.1.101/write2.php' --data-binary 'temperature=20.230'`

**Reply by paulzhdanov** [reply]
By the way, GET is also not working, but in a strange way, this
`      "0": { "h": [{ "h": "192.168.1.101", "u": "write2.php", "c": "var1=123&temperature=#HAA@0000" }] }
`
returns empty GET array, but this
`      "0": { "h": [{ "h": "192.168.1.101", "u": "write2.php?var1=123&temperature=#HAA@0000" }] }
`
doesn't ping the script at all, no calls in access log

@RavenSystem could you please check if everything' well with proper firmware module? Or may be there are any specific headers are required? I doubt, but still...

**Reply by RavenSystem** [reply]
Check HAA logs.

**Reply by paulzhdanov** [reply]
@RavenSystem this is the log for very first script and there is a call, but POST parameters are empty on a backend.
###
Aug 06 19:51:49 192.168.1.195-121A66 [6] New 192.168.1.181:8922 (2/20) Free HEAP 34644
Aug 06 19:51:49 192.168.1.195-121A66 [6] Verify 1/2
Aug 06 19:51:49 192.168.1.195-121A66 [6] Verify 2/2
Aug 06 19:51:49 192.168.1.195-121A66 [6] Pair found
Aug 06 19:51:50 192.168.1.195-121A66 [6] Verify OK
Aug 06 19:51:50 192.168.1.195-121A66 [6] Get Acc
Aug 06 19:51:52 192.168.1.195-121A66 <1> TH sensor
Aug 06 19:51:53 192.168.1.195-121A66 <1> TEMP 29.5C
Aug 06 19:51:53 192.168.1.195-121A66 <1> Wilcard 0 0.00
Aug 06 19:51:53 192.168.1.195-121A66 <1> Exec Act 51
Aug 06 19:51:53 192.168.1.195-121A66 <1> HUM 99
Aug 06 19:51:55 192.168.1.195-121A66 [6] Closing
Aug 06 19:51:55 192.168.1.195-121A66 [6] Closed (1/20)
Aug 06 19:51:55 192.168.1.195-121A66 [6] New 192.168.1.103:749 (2/20) Free HEAP 31068
Aug 06 19:51:55 192.168.1.195-121A66 [6] Verify 1/2
Aug 06 19:51:56 192.168.1.195-121A66 <1> Payload:
Aug 06 19:51:56 192.168.1.195-121A66 POST /write2.php HTTP/1.1
Aug 06 19:51:56 192.168.1.195-121A66 [6] Verify 2/2
Aug 06 19:51:56 192.168.1.195-121A66 [6] Pair found
Aug 06 19:51:57 192.168.1.195-121A66 [7] Verify 2/2
Aug 06 19:51:57 192.168.1.195-121A66 [7] Pair found
Aug 06 19:51:57 192.168.1.195-121A66 [7] Verify OK
Aug 06 19:51:57 192.168.1.195-121A66 [7] Get Acc
Aug 06 19:51:57 192.168.1.195-121A66 [8] New 192.168.1.204:38339 (4/20) Free HEAP 27012
Aug 06 19:51:57 192.168.1.195-121A66 [8] Verify 1/2
Aug 06 19:51:58 192.168.1.195-121A66 [8] Verify 2/2
Aug 06 19:51:58 192.168.1.195-121A66 [8] Pair found
Aug 06 19:51:59 192.168.1.195-121A66 [8] Verify OK
Aug 06 19:51:59 192.168.1.195-121A66 [6] Get Acc
Aug 06 19:51:59 192.168.1.195-121A66 [8] Get Acc
Aug 06 19:52:02 192.168.1.195-121A66 <1> TH sensor
Aug 06 19:52:03 192.168.1.195-121A66 <1> TEMP 27.2C
Aug 06 19:52:03 192.168.1.195-121A66 <1> Wilcard 0 0.00
Aug 06 19:52:03 192.168.1.195-121A66 <1> Exec Act 51
Aug 06 19:52:03 192.168.1.195-121A66 <1> HUM 99
Aug 06 19:52:03 192.168.1.195-121A66 [5] Send Ev
Aug 06 19:52:06 192.168.1.195-121A66 <1> Payload:
Aug 06 19:52:07 192.168.1.195-121A66 [6] Closing
Aug 06 19:52:07 192.168.1.195-121A66 [6] Closed (2/20)
Aug 06 19:52:12 192.168.1.195-121A66 <1> TH sensor
Aug 06 19:52:13 192.168.1.195-121A66 <1> TEMP 26.5C
Aug 06 19:52:13 192.168.1.195-121A66 <1> Wilcard 0 0.00
Aug 06 19:52:13 192.168.1.195-121A66 <1> Exec Act 51
Aug 06 19:52:13 192.168.1.195-121A66 <1> HUM 99
Aug 06 19:52:13 192.168.1.195-121A66 <1> Net Action 192.168.1.101:80
Aug 06 19:52:16 192.168.1.195-121A66 <1> Payload:
Aug 06 19:52:16 192.168.1.195-121A66 POST /write2.php HTTP/1.1
Aug 06 19:52:16 192.168.1.195-121A66 Host: 192.168.1.101
Aug 06 19:52:19 192.168.1.195-121A66 [5] Get Ch
Aug 06 19:52:19 192.168.1.195-121A66 [5] for 1.6
Aug 06 19:52:22 192.168.1.195-121A66 <1> TH sensor
Aug 06 19:52:23 192.168.1.195-121A66 <1> TEMP 26.1C
Aug 06 19:52:23 192.168.1.195-121A66 <1> Wilcard 0 0.00
Aug 06 19:52:23 192.168.1.195-121A66 <1> Exec Act 51
Aug 06 19:52:23 192.168.1.195-121A66 <1> HUM 99
Aug 06 19:52:23 192.168.1.195-121A66 [5] Send Ev
Aug 06 19:52:26 192.168.1.195-121A66 <1> Payload:
Aug 06 19:52:26 192.168.1.195-121A66 POST /write2.php HTTP/1.1
Aug 06 19:52:32 192.168.1.195-121A66 <1> TH sensor
Aug 06 19:52:33 192.168.1.195-121A66 <1> TEMP 25.8C
Aug 06 19:52:33 192.168.1.195-121A66 <1> Wilcard 0 0.00
Aug 06 19:52:33 192.168.1.195-121A66 <1> Exec Act 51
Aug 06 19:52:33 192.168.1.195-121A66 <1> HUM 99
Aug 06 19:52:36 192.168.1.195-121A66 <1> Payload:
Aug 06 19:52:36 192.168.1.195-121A66 POST /write2.php HTTP/1.1
Aug 06 19:52:36 192.168.1.195-121A66 Host: 192.168.1.101
Aug 06 19:52:42 192.168.1.195-121A66 <1> TH sensor
Aug 06 19:52:43 192.168.1.195-121A66 <1> TEMP 25.6C
Aug 06 19:52:43 192.168.1.195-121A66 <1> Wilcard 0 0.00
Aug 06 19:52:43 192.168.1.195-121A66 <1> Exec Act 51
Aug 06 19:52:43 192.168.1.195-121A66 <1> HUM 99
Aug 06 19:52:43 192.168.1.195-121A66 <1> Net Action 192.168.1.101:80
Aug 06 19:52:43 192.168.1.195-121A66 [5] Send Ev
Aug 06 19:52:45 192.168.1.195-121A66 <1> Payload:
Aug 06 19:52:45 192.168.1.195-121A66 POST /write2.php HTTP/1.1
Aug 06 19:52:45 192.168.1.195-121A66 Host: 192.168.1.101
Aug 06 19:52:45 192.168.1.195-121A66 User-Agent: HAA/10.8.3 esp8266
###

This script
`"0": { "h": [{ "h": "192.168.1.101", "u": "write2.php?var1=123&temperature=#HAA@0000" }] }`
doesn't work at all, device is not accessible right after load

**Reply by paulzhdanov** [reply]
@RavenSystem any advise?
